src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -318,24 +370,67 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
         }
     }
 
-    private long writePQ(SequentialWriter writer) throws IOException
+    // taken and adopted from com.github.jbellis.jvector.disk.OnDiskGraphIndex
+    private static <T> void writeGraph(GraphIndex<T> graph,
```

Comment:
i think it will be cleaner to implement a GraphIndex that does the remapping, so you don't have to change the write code itself

Reply:
I'd do it a separate set of changes (jvector's change with this + update to the tests etc), and then cassandra change with matching jvector release